Assets/Scripts/PoolSystem/PoolSystem.cs

```diff
@@ -0,0 +1,119 @@
+
+        /// <summary>
+        /// Creates new object if there aren't objects in the pool or put objects from the pool into the scene.
+        /// </summary>
+        /// <param name="nameTag">The name of the category the object is in</param>
+        /// <param name="parentTransform">The object you want the spawnobject to be the childobject of</param>
+        /// <param name="spawnPrefab">The object you want to spawn in</param>
+        public void SpawnObject(string nameTag, Transform parentTransform = null, GameObject spawnPrefab = null)
+        {
+            if (_objectsQueu.ContainsKey(nameTag) && _objectsQueu[nameTag].Count > 0)
+            {
+                List<GameObject> objects = _objectsQueu[nameTag];
+                int objectCount = objects.Count - 1;
+
+                MoveObjectToActiveScene(objects[objectCount], parentTransform);
+                AddActiveObject(nameTag, objects[objectCount]);
+
+                objects[objectCount].SetActive(true);
+                objects.RemoveAt(objectCount);
+                return;
+            }
+
+            if (spawnPrefab == null) return;
+
+            GameObject newGameObject = Instantiate(spawnPrefab);
+            MoveObjectToActiveScene(newGameObject, parentTransform);
+            AddActiveObject(nameTag, newGameObject);
+        }
+
+        /// <summary>
+        /// Puts the given object in the poolsystem
+        /// </summary>
+        /// <param name="nameTag">The name of the category the object should be in</param>
+        /// <param name="despawnObject">The object you want to despawn</param>
+        public void DespawnObject(string nameTag, GameObject despawnObject)
+        {
+            if (_objectsActive.ContainsKey(nameTag) && _objectsActive[nameTag].Contains(despawnObject))
+            {
+                List<GameObject> objects = _objectsActive[nameTag];
+
+                MoveObjectToDontDestroyOnLoadScene(despawnObject);
+                AddQueuObject(nameTag, despawnObject);
+
+                despawnObject.SetActive(false);
+                objects.Remove(despawnObject);
+                return;
+            }
+
+            Debug.LogError($"Can't find object {nameTag} to despawn");
+        }
+
+        /// <summary>
+        /// Starts coroutine DeathTimer
+        /// </summary>
+        /// <param name="amountOfTimer">How long do you want the death timer to last</param>
+        /// <param name="objectName">Name of the object you want to bring back after the timer</param>
+        /// <param name="previousParentTransform">The parent object you want the object to appear under</param>
+        public void StartDeathTimer(int amountOfTimer, string objectName, Transform previousParentTransform) 
+            => StartCoroutine(DeathTimer(amountOfTimer, objectName ,previousParentTransform));
+
+        private IEnumerator DeathTimer(int amountOfTimer, string objectName, Transform previousParentTransform)
```

Comment:
Could you rename the method, because when I think of a DeathTimer method I woudnt think it also respawns it.